bin/runtime-common/src/refund_relayer_extension.rs

```diff
@@ -265,46 +307,46 @@ where
 		_info: &DispatchInfoOf<Self::Call>,
 		_len: usize,
 	) -> TransactionValidity {
-		if let Some(calls) = self.expand_call(call) {
-			for nested_call in calls {
-				nested_call.check_obsolete_submit_finality_proof()?;
-				nested_call.check_obsolete_submit_parachain_heads()?;
-				nested_call.check_obsolete_call()?;
+		// this is the only relevant line of code for the `pre_dispatch`
+		//
+		// we're not calling `validato` from `pre_dispatch` directly because of performance
+		// reasons, so if you're adding some code that may fail here, please check if it needs
+		// to be added to the `pre_dispatch` as well
+		let parsed_call = self.parse_and_check_for_obsolete_call(call)?;
+
+		// the following code just plays with transaction priority and never returns an error
+		let mut valid_transaction = ValidTransactionBuilder::default();
+		if let Some(parsed_call) = parsed_call {
+			// we give delivery transactions some boost, that depends on number of messages inside
+			let messages_call_info = parsed_call.messages_call_info();
+			if let MessagesCallInfo::ReceiveMessagesProof(_) = messages_call_info {
+				// compute total number of messages in transaction
+				let bundled_messages = messages_call_info.bundled_messages();
+
+				// a quick check to avoid invalid high-priority transactions
+				if bundled_messages <= Runtime::MaxUnconfirmedMessagesAtInboundLane::get() {
```

Comment:
I think it would be good to have a test for this condition as well. Personally I didn't see one. I hope I didn't miss it.

Also in order to make it easier to test, maybe it would be better to move this check inside `compute_priority_boost` ?

Reply:
It should be tested indeed! Thanks for spotting that